kite-data/kite-data-spark/src/main/scala/org/kitesdk/data/spark/KiteDatasetSaver.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2015 Cloudera, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kitesdk.data.spark
+
+import java.net.URI
+
+import com.databricks.spark.avro.SchemaSupport
+import org.apache.avro.generic.GenericData.Record
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Row}
+import org.kitesdk.data._
+import org.kitesdk.data.mapreduce.DatasetKeyOutputFormat
+
+object KiteDatasetSaver extends SchemaSupport {
+
+  private def rowsToAvro(rows: Iterator[Row],
+                         schema: StructType): Iterator[(Record, Null)] = {
+    val converter = createConverter(schema, "topLevelRecord")
+    rows.map(x => (converter(x).asInstanceOf[Record], null))
+  }
+
+  private def saveAsKiteDataset(dataFrame: DataFrame,
+                                uri: URI,
+                                format: Format,
+                                compressionType: CompressionType,
+                                partitionStrategy: Option[PartitionStrategy],
+                                columnMapping: Option[ColumnMapping],
```

Comment:
I was thinking to pass a descriptor instead, The problem I found is that I have to specify everything in the descriptor including the schema, but the schema is generated internally from the passed data frame.

For writing I'm using the kite output format, so if it's possible to append using the kite output format I could do it as well.